pkg/deploy/registry/devfileregistry.go

```diff
@@ -0,0 +1,104 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package registry
+
+import (
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/eclipse-che/che-operator/pkg/deploy/expose"
+)
+
+type DevfileRegistry struct {
+	deployContext *deploy.DeployContext
+}
+
+func NewDevfileRegistry(deployContext *deploy.DeployContext) *DevfileRegistry {
+	return &DevfileRegistry{
+		deployContext: deployContext,
+	}
+}
+
+func (p *DevfileRegistry) SyncAll() (bool, error) {
+	done, err := p.SyncService()
+	if !done {
+		return false, err
+	}
+
+	endpoint, done, err := p.ExposeEndpoint()
+	if !done {
+		return false, err
+	}
+
+	done, err = p.UpdateStatus(endpoint)
+	if !done {
+		return false, err
+	}
+
+	done, err = p.SyncConfigMap()
+	if !done {
+		return false, err
+	}
+
+	done, err = p.SyncDeployment()
```

Comment:
When do we need to use separate methods instead of `SyncAll`?

Reply:
- it allows to test them separately if needed
- it looks clearer

Reply:
Also it makes think that it is ok to use, say, `SyncConfigMap` on its own